### PR TITLE
Add the ability to use Key/Value data with DataBuilders. Fixes #1093

### LINKED ISSUE #1093: Add data methods using keys and values to BlockState.Builder and similar classes.
Currently the builders for data-backed classes do not allow using the key value system, and instead require creation of an entire manipulator.

### PR DESCRIPTION
**Pull request by me4502**
This PR aims to allow the usage of Key/Value data with DataBuilders, such as BlockState.Builder, etc.

This fixes #1093.

### CODE REVIEW

File: src/main/java/org/spongepowered/api/item/inventory/ItemStack.java [hunk 1]
```diff
@@ -35,12 +32,17 @@
...
 import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.data.key.Key;
 import org.spongepowered.api.data.manipulator.DataManipulator;
 import org.spongepowered.api.data.manipulator.ImmutableDataManipulator;
+import org.spongepowered.api.data.value.BaseValue;
 import org.spongepowered.api.item.ItemType;
 import org.spongepowered.api.text.translation.Translatable;
 import org.spongepowered.api.util.ResettableBuilder;
 
+import static com.google.common.base.Preconditions.checkArgument;
```
Comment (gabizou) [inline src/main/java/org/spongepowered/api/item/inventory/ItemStack.java]: I'm pretty sure the static imports come first.

Comment (me4502) [inline src/main/java/org/spongepowered/api/item/inventory/ItemStack.java]: Yeah, I'll re-import the style thing.

**Comment by me4502** [comment]
@gabizou fixed

**Comment by gabizou** [comment]
Merged.

**Pull request closed without merging**